src/main/java/seedu/address/model/person/GradeList.java

```diff
@@ -175,7 +187,14 @@ public boolean equals(Object other) {
         }
 
         GradeList otherGradeList = (GradeList) other;
-        return this.grades.equals(otherGradeList.grades);
+        // Sort both grade lists before comparing
+        List<Grade> sortedGrades = new ArrayList<>(this.grades.values());
+        sortedGrades.sort(Comparator.comparing(Grade::getTestName));
+
+        List<Grade> otherSortedGrades = new ArrayList<>(otherGradeList.grades.values());
+        otherSortedGrades.sort(Comparator.comparing(Grade::getTestName));
```

Comment:
simply comparing does not work. For example:

`[a, B]`
and
`[A,b]`

Instead, you can try to convert everything to lower case first before sorting and comparing.